src/test/java/uk/gov/ida/matchingserviceadapter/services/EidasMatchingRequestToLMSRequestTransformTest.java

```diff
@@ -0,0 +1,115 @@
+import static uk.gov.ida.matchingserviceadapter.services.AttributeStatementBuilder.aCurrentFamilyNameAttribute;
+import static uk.gov.ida.matchingserviceadapter.services.AttributeStatementBuilder.aCurrentGivenNameAttribute;
+import static uk.gov.ida.matchingserviceadapter.services.AttributeStatementBuilder.aDateOfBirthAttribute;
+import static uk.gov.ida.matchingserviceadapter.services.AttributeStatementBuilder.aGenderAttribute;
+import static uk.gov.ida.matchingserviceadapter.services.AttributeStatementBuilder.aPersonIdentifierAttribute;
+import static uk.gov.ida.matchingserviceadapter.services.AttributeStatementBuilder.aPlaceOfBirthAttribute;
+import static uk.gov.ida.matchingserviceadapter.services.AttributeStatementBuilder.anEidasAttributeStatement;
+import static uk.gov.ida.saml.core.test.builders.AssertionBuilder.anAssertion;
+import static uk.gov.ida.saml.core.test.builders.AuthnContextBuilder.anAuthnContext;
+import static uk.gov.ida.saml.core.test.builders.AuthnContextClassRefBuilder.anAuthnContextClassRef;
+import static uk.gov.ida.saml.core.test.builders.AuthnStatementBuilder.anAuthnStatement;
+
+@RunWith(OpenSAMLMockitoRunner.class)
+public class EidasMatchingRequestToLMSRequestTransformTest {
+    public static final org.joda.time.LocalDate DOB = org.joda.time.LocalDate.parse("2001-02-01", ISODateTimeFormat.dateTimeParser());
+    private Assertion assertion;
+
+    private EidasMatchingRequestToLMSRequestTransform transform;
+
+    @Mock
+    private AttributeQuery attributeQuery;
+
+    @Mock
+    private UserIdHashFactory pidHashFactory;
+
+    @Before
+    public void setUp() {
+        when(attributeQuery.getID()).thenReturn("the-aqr-id");
+        assertion = anAssertion()
+            .addAuthnStatement(
+                anAuthnStatement()
+                    .withAuthnContext(anAuthnContext()
+                        .withAuthnContextClassRef(
+                            anAuthnContextClassRef()
+                            .withAuthnContextClasRefValue(LevelOfAssurance.SUBSTANTIAL.toString())
+                            .build()
+                        ).build()
+                    )
+                .build()
+            )
+            .addAttributeStatement(
+                anEidasAttributeStatement(
+                    aCurrentGivenNameAttribute("Fred"),
+                    aCurrentFamilyNameAttribute("Flintstone"),
+                    aPersonIdentifierAttribute("the-pid"),
+                    aGenderAttribute("MALE"),
+                    aDateOfBirthAttribute(DOB),
+                    aBirthNameAttribute("birth-name"),
+                    aPlaceOfBirthAttribute("place-of-birth")
+                ).build()
+            ).buildUnencrypted();
+        transform = new EidasMatchingRequestToLMSRequestTransform(pidHashFactory);
+        when(pidHashFactory.hashId(anyString(), anyString(), any())).thenReturn("the-hashed-pid");
```

Comment:
Does it make sense to mock hashId only for the correct input here instead of using 'any' matchers?